[#169] Style: 짤 업로드 페이지 UI 수정

HY219 opened the pull request:
## 📝 작업 내용

- 전체 화면 요소를 중앙에 배치
- 필수 항목 표시 아이콘 추가
  - `import { Asterisk } from "lucide-react";`
- UploadGuide 컴포넌트 삭제
- 작은 화면에서 input 창이 잘리는 이슈 -> mx-auto & flow-wrap 적용하여 해결


## 📷 스크린샷 (선택)
![image](https://github.com/zzalmyu/zzalmyu-frontend/assets/82134668/17dc4a7f-917b-4e7b-b380-82d79edae7e5)

## 💬 리뷰 요구사항(선택)
- 제일 밖에 감싸고 있는 div의 css 속성은 그대로 유지하고, 전체 css 수정해주었습니다.
- "짤 업로드" 페이지 제목 아래에 선을 추가해준다면, 다른 페이지에서도 동일하게 적용해주어야할 것 같은데 어떻게 생각하시나요. 공통으로 추가하는 것이 좋을까요?
- 기기별 화면 대응
  - 더 좋은 방법이 있다면 알려주세요!
  - 반응형으로 breakpoint를 두어 input창이 줄어들도록 작성해주고 싶었는데, Minimum width를 사용자 지정해주려면 tailwind.config 파일에 추가해줘야해서, 다른 화면에도 영향이 가게 되어 이 방법은 사용하지 않았습니다.
  - 요소를 가운데 배치하기 위해 mx-auto || justify-center를 사용하면, margin이 자동 지정되어서 input창 크기 조절이 불가능했고 || 요소가 창을 벗어나는 경우가 생겨서 화면 대응이 불가능했습니다. -> 그래서 mx-auto와 함께 flow-wrap 사용


## 📍 기타 (선택)
* 용재님께서 최종 UI 수정하시면서 수정사항 있으면 같이 수정하신다하셔서, 큰 이상 없으면 approve 해도 좋을 것 같습니다:)

close #169

HY219 committed:
Style: 짤 업로드 전체 UI 수정

중앙 배치

HY219 committed:
Style: 필수 입력 항목 표시 아이콘 추가

lucide에서 Asterisk 아이콘 사용

HY219 committed:
Merge branch 'main' of https://github.com/zzalmyu/zzalmyu-frontend into #169/style/upload-page-ui

HY219 committed:
Cleanup: 오입력된 부분 제거

hyeonjinan096 commented on src/routes/_authentication/upload-zzal/index.tsx:
```diff
@@ -83,32 +83,64 @@ const UploadZzal = () => {
-        <UploadGuide />
-        <div className="mt-20pxr flex w-full flex-col sm:flex-row sm:items-start">
-          <div className="mx-auto flex w-320pxr flex-col">
+      <div className="flex flex-col justify-center gap-20pxr px-50pxr pt-30pxr sm:px-100pxr">
```
pt-30pxr top에만 패딩을 줘서인지 큰 모니터에서는 괜찮은데 노트북 화면으로는 요소가 화면 밑에 붙어 있습니다. p-30pxr로 수정해야할 것 갔습니다:)
![image](https://github.com/zzalmyu/zzalmyu-frontend/assets/107539614/d5150d4a-6f10-4fa9-bb73-2e8951e8c29a)

hyeonjinan096 commented on src/routes/_authentication/upload-zzal/index.tsx:
```diff
@@ -83,32 +83,64 @@ const UploadZzal = () => {
-                value={imageTitle}
-                className="z-20 min-h-12 flex-1 rounded-xl border-none bg-transparent outline-none"
-              />
+          <div className="mt-10 flex h-536pxr w-500pxr flex-col sm:ml-36 sm:mt-0 sm:justify-between">
```
div를 w-500pxr로 고정해서인지 내부 검색창이나 버튼 반응형 동작이 안되는 거 같습니다ㅠ

HY219 replied:
일부로 움직이지 않도록 고정해줬는데, 작은 화면에서 검색창이 잘리나보네요 😢  수정하도록하겠습니다!

hyeonjinan096 reviewed (commented):
혜연님 수고하셨습니다:) 큰 화면에서는 가운데로 잘 보이는데 모니터 사이즈에 따라 위치가 달라지는 거 같습니다. width를 고정값으로 지정해서 일 것 같은데 정확한 원인은 모르겠네요ㅠ 확인해보시면 좋을 것 같습니다.

choi-ik reviewed (commented):
<img width="758" alt="image" src="https://github.com/zzalmyu/zzalmyu-frontend/assets/91654577/b9d805d4-2616-4b57-98cf-0d6f258ab47e">
반응형이 특정 breakPoint에서만 작동하고, 화면 사이즈가 줄어듬에 따라 input의 가로 길이가 줄어들어거나 하는 반응형 로직이 없는데 이 부분도 추가가되면 좋을 것 같네요 ㅎ

HY219 committed:
Style: 검색 창이 화면에서 가릴 때 세로 보기로 UI 변경

flex-wrap
중앙정렬을 위해 mx-auto || justify-center 속성을 적용할 경우 화면에 따라 크기 대응이 안되는 이슈로 flex-wrap을 사용해 주었습니다.

HY219 committed:
Style: 업로드할 파일과 input 사이의 pxr 수정

sm:24pxr

yjc2021 reviewed (approved):
LGTM

choi-ik reviewed (approved):
LGTM!

HY219 committed:
Merge branch 'main' into #169/style/upload-page-ui

thalassophilia commented:
에러 메시지입니다!
<img width="1210" alt="스크린샷 2024-03-21 오후 8 07 04" src="https://github.com/zzalmyu/zzalmyu-frontend/assets/61978339/367a1a3d-6d29-41ee-884d-a683f68b999b">

hyeonjinan096 reviewed (approved):
LGTMLGTM

HY219 committed:
[#181] Modify : 짤 업로드 페이지에 TagSlider 컴포넌트 적용 (#182)

Modify: 기존 selectedTag를 보여주던 부분을 TagSlider로 대체

Modify: input 창에 태그 입력 후 reset 

Refactor: UploadTagSlider 삭제 및 공통 TagBadge, TagSlider 컴포넌트 사용 

Style: TagSlider 영역의 width 수정

Rename: selectedTag를 selectedUploadTag로 변경

Modify: 태그 input 글자 수 제한 

Fix: tag 입력시 input 초기화

Rename: newTagId, newTagName을 uploadTagId, uploadTagName으로 변경
